include/eve/module/core/function/simd/x86/min.hpp

```diff
@@ -10,42 +10,52 @@
 //==================================================================================================
 #pragma once
 
-#include <eve/detail/overload.hpp>
+#include <eve/concept/value.hpp>
 #include <eve/detail/abi.hpp>
+#include <eve/detail/overload.hpp>
 #include <eve/forward.hpp>
+#include <eve/function/if_else.hpp>
 #include <type_traits>
-#include <eve/concept/value.hpp>
 
 namespace eve::detail
 {
+  template <eve::simd_value Wide>
+  Wide if_else_min(Wide x, Wide y)
+  {
+    return eve::if_else(x > y, y, x);
+  }
+
   // -----------------------------------------------------------------------------------------------
   // 128 bits implementation
   template<real_scalar_value T, typename N>
   EVE_FORCEINLINE wide<T, N, x86_128_> min_(EVE_SUPPORTS(sse2_)
                                        , wide<T, N, x86_128_> const &v0
                                        , wide<T, N, x86_128_> const &v1) noexcept
   {
-         if constexpr(std::is_same_v<T, double>)   return _mm_min_pd(v0, v1);
-    else if constexpr(std::is_same_v<T, float>)    return _mm_min_ps(v0, v1);
-    else if constexpr( std::is_integral_v<T>)
+         if constexpr(std::is_same_v<T, double>) return _mm_min_pd(v0, v1);
+    else if constexpr(std::is_same_v<T, float>)  return _mm_min_ps(v0, v1);
+    else if constexpr(sizeof(T) == 1)
     {
-      constexpr bool issigned = std::is_signed_v<T>;
-           if constexpr(issigned && sizeof(T) == 2) return _mm_min_epi16(v0, v1);
-      else if constexpr(!issigned && sizeof(T) == 1) return _mm_min_epu8(v0, v1);
-      else if constexpr((current_api >= sse4_1) && (sizeof(T) != 8))
-      {
-        if constexpr(issigned)
-        {
-               if constexpr(sizeof(T) == 1) return _mm_min_epi8(v0, v1);
-          else if constexpr(sizeof(T) == 4) return _mm_min_epi32(v0, v1);
-        }
-        else
-        {
-               if constexpr(sizeof(T) == 2)  return _mm_min_epu16(v0, v1);
-          else if constexpr(sizeof(T) == 4)  return _mm_min_epu32(v0, v1);
-        }
-      }
-      else return map(min, v0, v1);
+           if constexpr(!std::is_signed_v<T>)    return _mm_min_epu8(v0, v1);
```

Comment:
could this case and the 16 bits case be in the same order (is_signed then > sse4.2) ?